api.go

```diff
@@ -14,24 +15,64 @@ func getRouter(db *gorm.DB, p providers.StorageProvider) *mux.Router {
 
 	f := r.PathPrefix("/files").Subrouter()
 	f.HandleFunc("/{name}", uploadFile(db, p)).Methods("POST")
+	f.HandleFunc("/{name}", deleteFile(db, p)).Methods("DELETE")
 
 	return r
 }
 
+func deleteFile(db *gorm.DB, p providers.StorageProvider) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		name := vars["name"]
+		file := model.File{}
+		err := db.Where("name = ?", name).First(&file).Error
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusNotFound)
+			return
+		}
+
+		if err := p.DeleteFile(name); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+		}
+
+		err = db.Model(&model.File{}).Delete(&model.File{}, name).Error
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		w.WriteHeader(http.StatusPartialContent)
+		return
+	}
+}
+
 func uploadFile(db *gorm.DB, p providers.StorageProvider) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 		vars := mux.Vars(r)
 		name := vars["name"]
 		data := r.Body
-		if r.ContentLength < 1 {
-			http.Error(w, "ContentLength is not set!", http.StatusBadRequest)
+
```

Comment:
We need to handle both multipart/form-data and direct upload. Distinguishing between those two could be separated in two different methods (eg. uploadFileData and uploadFileMultipart) and the uploadFile function can check the content type if it’s multipart/form-data to handle it to uploadFileMultipart or to uploadFileData otherwise (all requests with content-type != multipart/form-data)